library/Zend/Code/Scanner/ClassScanner.php

```diff
@@ -480,6 +505,142 @@ public function hasProperty($name)
         return false;
     }
 
+
+    public function getTraits()
+    {
+        if(empty($this->traits)) {
+            //get list of traitNames
+            $traitNames = $this->getTraitNames();
+            foreach($traitNames as $traitName) {
+                /*
+                 * @todo test if this works when trait is included or required instead of autoloaded?
+                 */
+                $fileName = str_replace("\\", "/", $traitName) .".php";
```

Comment:
Please be aware that changing this to not using reflection could be a BC-break depending on how it is changed. Code can depend on the fact that the traits are autoloaded by `ReflectionClass`.